server/controllers/temporary-accommodation/manage/turnaroundsController.ts

```diff
@@ -0,0 +1,78 @@
+import type { Request, RequestHandler, Response } from 'express'
+
+import type { NewTurnaround } from '@approved-premises/api'
+import paths from '../../../paths/temporary-accommodation/manage'
+import { BedspaceService, BookingService, PremisesService, TurnaroundService } from '../../../services'
+import { generateTurnaroundConflictBespokeError } from '../../../utils/bookingUtils'
+import { parseNaturalNumber } from '../../../utils/formUtils'
+import extractCallConfig from '../../../utils/restUtils'
+import {
+  catchValidationErrorOrPropogate,
+  fetchErrorsAndUserInput,
+  insertBespokeError,
+  insertGenericError,
+} from '../../../utils/validation'
+
+export default class TurnaroundsController {
+  constructor(
+    private readonly premisesService: PremisesService,
+    private readonly bedspaceService: BedspaceService,
+    private readonly bookingService: BookingService,
+    private readonly turnaroundService: TurnaroundService,
+  ) {}
+
+  new(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const { errors, errorSummary, errorTitle, userInput } = fetchErrorsAndUserInput(req)
+      const { premisesId, roomId, bookingId } = req.params
+
+      const callConfig = extractCallConfig(req)
+
+      const premises = await this.premisesService.getPremises(callConfig, premisesId)
+      const room = await this.bedspaceService.getRoom(callConfig, premisesId, roomId)
+      const booking = await this.bookingService.getBooking(callConfig, premisesId, bookingId)
+
+      return res.render('temporary-accommodation/turnarounds/new', {
+        premises,
+        room,
+        booking,
+        errors,
+        errorSummary,
+        errorTitle,
+        workingDays: `${booking.turnaround?.workingDays || 0}`,
```

Comment:
I'm a bit confused by this being handed in to the new turnaround template - how is it being used?

Reply:
Ah, this is just prepopulating the field with the current turnaround days (so from the user's PoV it's just editing the turnaround)

Reply:
Discussed this one with Dan - I'd hoped I could rely on a turnaround always being there, but sounds like there's a possibility that some bookings escaped the migration. If it's not there, it'll be treated by the API as a zero-day turnaround and I'm following that UI side

Reply:
ah ok, makes sense!